jhputnam/test_mailroom.py

```diff
@@ -0,0 +1,140 @@
+"""
+Mailroom testing.
+"""
+
+import pickle
+import pytest
+
+import mailroom
+
+
+def test_initdb_dbpresent():
+    """Test dbinit with db file present."""
+
+    testdb = {'Dan Rather': [3453.0], 'SpongeBob': [44.05]}
+
+    # Call initdb with test data.
+    donors = mailroom.initdb('testdb.dat')
+
+    # Assert that the test data is what we expect.
+    assert donors == testdb
+
+
+def test_initdb_dbmissing1(monkeypatch):
+    """Test dninit function with db missing and option 1."""
+
+    # Override input and supply option 1.
+    monkeypatch.setitem(__builtins__, 'input', lambda x: '1')
```

Comment:
I believe it was a combo of the standard monkeypatch docs: http://doc.pytest.org/en/latest/monkeypatch.html

And a few stackoverflow threads such as this one:
http://stackoverflow.com/questions/35851323/pytest-how-to-test-a-function-with-input-call

Seemed pretty straightforward, and it's nice that it doesn't require any manual cleanup.